pkg/common/selector/selector.go

```diff
@@ -26,3 +33,13 @@ func (s *Selector) Raw() *common.Selector {
 	}
 	return c
 }
+
+func Validate(s *common.Selector) error {
+	// Validate that the Type does not contain a colon (:) to prevent accidental misconfigurations
+	// e.g. type="unix:user" value="root" is the invalid selector
+	// and type="unix" value"user:root" is the valid selector
+	if strings.Contains(s.Type, Delimiter) {
+		return fmt.Errorf("type in selector contains a colon, invalid type: '%s'", s.Type)
```

Comment:
small nit on the error message here since it will eventually bubble back up to an API consumer. Would be great if we can be really clear on what is not allows, something like
```suggestion
		return fmt.Errorf("selector type must not contain a colon; invalid selector type: %q", s.Type)
```